File: ios/Classes/teb/MyBoardCallback.swift
//
// Created by vicky Leu on 2021/5/19.
//

import Foundation

public class  MyBoardCallback :NSObject, TEduBoardDelegate{

     let TAG = "MyBoardCallback"
    
    
    let boardAwareCallback:BoardAwareInterface
    
    init(_ boardAwareCallback: BoardAwareInterface) {
        self.boardAwareCallback=boardAwareCallback
    }

    public func onTEBError(_ code: TEduBoardErrorCode, msg: String!) {
        print("\(TAG) onTEBError:\(code)|\(msg)")
    }
    
    public func onTEBWarning(_ code: TEduBoardWarningCode, msg: String!) {
        print("\(TAG) onTEBWarning:\(code)|\(msg)")
    }
    
    public func onTEBInit() {
        boardAwareCallback.addBoardView()
    }
    
    public func onTEBHistroyDataSyncCompleted() {
        boardAwareCallback.onTEBHistroyDataSyncCompleted()
    }
    
    public func onTEBSyncData(_ data: String!) {
        
    }
    
    public func onTEBUndoStatusChanged(_ canUndo: Bool) {
        boardAwareCallback.setCanUndo(canUndo)
    }
    
    public func onTEBRedoStatusChanged(_ canRedo: Bool) {
        boardAwareCallback.setCanRedo(canRedo)
    }
    
    public func onTEBImageStatusChanged(_ boardId: String!, url: String!, status: TEduBoardImageStatus) {
        print("\(TAG) onTEBImageStatusChanged:\(boardId)|\(url)|\(status)")
    }
    
    public func onTEBSetBackgroundImage(_ url: String!) {
        print("\(TAG) onTEBSetBackgroundImage:\(url)")
    }
    
    public func onTEBAddImageElement(_ url: String!) {
        print("\(TAG) onTEBAddImageElement:\(url)")
    }
    
    public func onTEBAddElement(_ elementId: String!, url: String!) {
    }
    
  
    
    public func onTEBBackgroundH5StatusChanged(_ boardId: String!, url: String!, status: TEduBoardBackgroundH5Status) {
        print("\(TAG) onTEBBackgroundH5StatusChanged:\(boardId)|\(url)|\(status)")
    }
    
    public func onTEBAddBoard(_ boardIds: [Any]!, fileId: String!) {
        print("\(TAG) onTEBAddBoard:\(boardIds)|\(fileId)")
    }
    
    public func onTEBDeleteBoard(_ boardIds: [Any]!, fileId: String!) {
        print("\(TAG) onTEBDeleteBoard:\(boardIds)|\(fileId)")
    }
    
    public func onTEBGotoBoard(_ boardId: String!, fileId: String!) {
        print("\(TAG) onTEBGotoBoard:\(boardId)|\(fileId)")
    }
    
    public func onTEBGotoStep(_ currentStep: UInt32, totalStep: UInt32) {
        print("\(TAG) onTEBGotoStep:\(currentStep)|\(totalStep)")
    }
    
    public func onTEBRectSelected() {
        print("\(TAG) onTEBRectSelected")
    }
    
    public func onTEBRefresh() {
        print("\(TAG) onTEBRefresh")
    }
    
    public func onTEBSnapshot(_ path: String!, errorCode code: TEduBoardErrorCode, errorMsg msg: String!) {
        print("\(TAG) onTEBSnapshot:\(path)|\(code)|\(msg)")
    }
    
    public func onTEBFileTranscodeProgress(_ result: TEduBoardTranscodeFileResult!, path: String!, errorCode: String!, errorMsg: String!) {
    }
    
    public func onTEBAddTranscodeFile(_ fileId: String!) {
        print("\(TAG) onTEBAddTranscodeFile:\(fileId)")
    }
    
    public func onTEBDeleteFile(_ fileId: String!) {
    }
    
    public func onTEBSwitchFile(_ fileId: String!) {
    }
    
    public func onTEBFileUploadProgress(_ path: String!, currentBytes: Int32, totalBytes: Int32, uploadSpeed: Int32, percent: Float) {
        print("\(TAG) onTEBFileUploadProgress:\(path)|\(currentBytes)|\(totalBytes)|\(uploadSpeed)|\(percent)")
    }
    
    public func onTEBFileUploadStatus(_ path: String!, status: TEduBoardUploadStatus, errorCode: Int32, errorMsg: String!) {
        print("\(TAG) onTEBFileUploadStatus:\(path)|\(status)|\(errorCode)|\(errorMsg)")
    }
    
    public func onTEBH5FileStatusChanged(_ fileId: String!, status: TEduBoardH5FileStatus) {
    }
    
    public func onTEBVideoStatusChanged(_ fileId: String!, status: TEduBoardVideoStatus, progress: CGFloat, duration: CGFloat) {
        print("\(TAG) onTEBVideoStatusChanged:\(fileId)|\(status)|\(progress)|\(duration)")
    }
    
    public func onTEBAudioStatusChanged(_ elementId: String!, status: TEduBoardAudioStatus, progress: CGFloat, duration: CGFloat) {
        print("\(TAG) onTEBAudioStatusChanged:\(elementId)|\(status)|\(progress)|\(duration)")
    }
    
    public func onTEBAddImagesFile(_ fileId: String!) {
        print("\(TAG) onTEBAddImagesFile:\(fileId)")
        _ = boardAwareCallback.addFile(fileId)
    }
    
    public func onTEBH5PPTStatusChanged(_ fileId: String!, status: TEduBoardH5PPTStatus, message: String!) {}
    

    // # onTEBDeleteElement
    public func onTEBRemoveElement(_ elementIds: [Any]!) {
    }
}

File: android/src/main/kotlin/com/bond/whiteboard/teb/MyBoardCallback.kt
package com.bond.whiteboard.teb

import android.util.Log
import com.tencent.rtmp.TXLog
import com.tencent.teduboard.TEduBoardController.TEduBoardCallback
import com.tencent.teduboard.TEduBoardController.TEduBoardTranscodeFileResult

class MyBoardCallback(var boardAwareCallback: BoardAwareInterface) : TEduBoardCallback {
    override fun onTEBError(code: Int, msg: String) {
        TXLog.i(TAG, "onTEBError:$code|$msg")
    }

    override fun onTEBWarning(code: Int, msg: String) {
        TXLog.i(TAG, "onTEBWarning:$code|$msg")
    }

    override fun onTEBInit() {
        boardAwareCallback.addBoardView()
    }

    override fun onTEBHistroyDataSyncCompleted() {
        boardAwareCallback.onTEBHistroyDataSyncCompleted()
    }

    override fun onTEBSyncData(data: String) {}
    override fun onTEBAddBoard(boardId: List<String>, fileId: String) {
        TXLog.i(TAG, "onTEBAddBoard:$fileId")
    }

    override fun onTEBDeleteBoard(boardId: List<String>, fileId: String) {
        TXLog.i(TAG, "onTEBDeleteBoard:$fileId|$boardId")
    }

    override fun onTEBGotoBoard(boardId: String, fileId: String) {
        TXLog.i(TAG, "onTEBGotoBoard:$fileId|$boardId")
    }

    override fun onTEBGotoStep(currentStep: Int, total: Int) {
        TXLog.i(TAG, "onTEBGotoStep:$currentStep|$total")
    }

    override fun onTEBRectSelected() {
        TXLog.i(TAG, "onTEBRectSelected:")
    }

    override fun onTEBRefresh() {
        TXLog.i(TAG, "onTEBRefresh:")
    }

    override fun onTEBDeleteFile(fileId: String) {}
    override fun onTEBSwitchFile(fileId: String) {}
    override fun onTEBAddTranscodeFile(s: String) {
        TXLog.i(TAG, "onTEBAddTranscodeFile:$s")
    }

    override fun onTEBUndoStatusChanged(canUndo: Boolean) {
        boardAwareCallback.setCanUndo(canUndo)
    }

    override fun onTEBRedoStatusChanged(canredo: Boolean) {
        boardAwareCallback.setCanRedo(canredo)
    }

    override fun onTEBFileUploadProgress(path: String, currentBytes: Int, totalBytes: Int, uploadSpeed: Int, percent: Float) {
        TXLog.i(TAG, "onTEBFileUploadProgress:$path percent:$percent")
    }

    override fun onTEBFileUploadStatus(path: String, status: Int, code: Int, statusMsg: String) {
        TXLog.i(TAG, "onTEBFileUploadStatus:$path status:$status")
    }

    override fun onTEBFileTranscodeProgress(s: String, s1: String, s2: String, tEduBoardTranscodeFileResult: TEduBoardTranscodeFileResult) {}
    override fun onTEBH5FileStatusChanged(fileId: String, status: Int) {}
    override fun onTEBAddImagesFile(fileId: String) {
        Log.i(TAG, "onTEBAddImagesFile:$fileId")
        val fileInfo = boardAwareCallback.addFile(fileId)
    }

    override fun onTEBVideoStatusChanged(fileId: String, status: Int, progress: Float, duration: Float) {
        Log.i(TAG, "onTEBVideoStatusChanged:$fileId | $status|$progress")
    }

    override fun onTEBAudioStatusChanged(elementId: String, status: Int, progress: Float, duration: Float) {
        Log.i(TAG, "onTEBAudioStatusChanged:$elementId | $status|$progress")
    }

    override fun onTEBSnapshot(path: String, code: Int, msg: String) {
        Log.i(TAG, "onTEBSnapshot:$path | $code|$msg")
    }

    override fun onTEBH5PPTStatusChanged(statusCode: Int, fid: String, describeMsg: String) {}
    override fun onTEBTextComponentStatusChange(status: String, id: String, value: String, left: Int, top: Int) {
        Log.e(TAG, "onTEBTextComponentStatusChange textH5Status:$status textH5Id:$id")
        boardAwareCallback.onTextComponentStatusChange(id, status)
    }

    override fun onTEBImageStatusChanged(boardId: String, url: String, status: Int) {
        TXLog.i(TAG, "onTEBImageStatusChanged:$boardId|$url|$status")
    }

    override fun onTEBSetBackgroundImage(url: String) {
        Log.i(TAG, "onTEBSetBackgroundImage:$url")
    }

    override fun onTEBAddImageElement(url: String) {
        Log.i(TAG, "onTEBAddImageElement:$url")
    }

    override fun onTEBAddElement(id: String, url: String) {}
    override fun onTEBDeleteElement(id: List<String>) {}
    override fun onTEBBackgroundH5StatusChanged(boardId: String, url: String, status: Int) {
        Log.i(TAG, "onTEBBackgroundH5StatusChanged:$boardId url:$boardId status:$status")
    }

    companion object {
        private const val TAG = "MyBoardCallback"
    }
}